lib/backend/k8s/resources/customresource.go

```diff
@@ -60,6 +60,8 @@ func (c *customK8sResourceClient) Create(ctx context.Context, kvp *model.KVPair)
 
 	// Convert the KVPair to the K8s resource.
 	resIn, err := c.convertKVPairToResource(kvp)
+	// Clean out the annotations for the resource given since that's only used for our internal storage.
+	defer CleanAnnotations(resIn)
```

Comment:
I don't think this ClearAnnotations belongs here.  Everything that needs doing should be in the convertKVPairToResource.

Reply:
And in this case it feels particularly strange to use defer to do this since you are assuming that resIn is what we return below - which makes things hard to maintain.  This would be better handled just before we return the result.

In any case - the convertResourceToKVPair should take care of the converting back to the Calico ObjectMeta.

Reply:
I have it deferred in these functions because I noticed that the input data to the functions was being modified. Since the `Object` interface we're dealing with doesn't have a `DeepCopy` method defined on it, it makes it difficult to modify the contents of the metadata without modifying the input data. I thought it would be cleaner to not let input data to functions modify the input data but if we're always expecting these functions and their outputs to be used in a controlled manner, then I don't think it matters.

Reply:
I have a couple of concerns.  Firstly doing this in a defer feels odd to me because it makes the flow of the code hard to follow.  It's clearer if you Modify the ObjectMeta, send the request, Update the ObjectMeta from the response data.  So really the use of defer is more that it's difficult to follow the flow here.  Is your concern that in the error case we've modified the resource that was given to us?  I thought DeepCopy was on the interface now?

Reply:
Yup, my concern was that in the error case (or any case without that function running, I had it in a `defer` just to make sure it was kicked off) we are modifying the input data. I see, I got the `runtime.Object` interface mixed up with the `metav1.Object` interface. I'll change this back.